### PR TITLE
[battery_plus] Migrate Android part to Kotlin

### PR DESCRIPTION
**Pull request by vbuberen**
## Description

One more PR in a series of migrating all Plus plugins to Kotlin. Saved existing behavior and tried to make the code easier to read.

Modernised the project to use latest Android dependencies and bumped targetSDK and compileSDK to 33. Nothing breaking  is expected.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        val status = getBatteryState()
+        publishBatteryStatus(events, status)
+    }
+
+    override fun onCancel(arguments: Any) {
+        applicationContext!!.unregisterReceiver(chargingStateChangeReceiver)
+        chargingStateChangeReceiver = null
+    }
+
+    private fun getBatteryState(): String? {
```
Comment (nohli) [inline packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt:91]: Is there a reason, this was renamed from `getBatteryStatus()` to `getBatteryState()`?
In line 59 it says `val currentBatteryStatus = getBatteryState()`, the method call is named `getBatteryState`.
Could it be `state` or `status` (breaking) everywhere?

**Review by nohli — APPROVED** [review]
LGTM!

Great work 💪🏽

File: packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt (hunk 1 above)
Comment (nohli) [reply packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt:91]: Could we rename everything containing `batteryStatus` to `batteryState`?

Comment (vbuberen) [reply packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt]: I have brought back the old name to this function for consistency with other functions and constants from the `BatteryManager` class, like `BatteryManager.BATTERY_STATUS_CHARGING`

Comment (nohli) [reply packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt]: Sounds and looks great 🚀

**Review by miquelbeltran — APPROVED** [review]
Thank you!

**Comment by miquelbeltran** [comment]
published

**Pull request merged**